app/helpers/matches_helper.rb

```diff
@@ -1,13 +1,8 @@
 module MatchesHelper
   def participant_link(participant)
     if participant.present?
-      participant_name = participant.name
-      if participant.is_a? Match
-        link_name = I18n.t 'matches.winner_of_match', match: participant_name
-        link_to link_name, match_path(participant)
-      else
-        link_to participant_name, team_path(participant)
-      end
+      path = participant.is_a?(MatchResult) ? match_path(participant.match) : team_path(participant)
```

Comment:
What if the outcome of the match is already known?

It looks like it will still link to the matches page, even though it should link  to the winning team (the displayed name will be the name of the winning team).